x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -644,6 +644,132 @@ public void testErrorMessageForInvalidIntervalParams() throws IOException {
+        Map<String, String> colA = Map.of("name", "boolean", "type", "boolean");
+        Map<String, String> colB = Map.of("name", "xx2", "type", "double");
+        assertEquals(List.of(colA, colB), result.get("columns"));
+        assertEquals(List.of(List.of(false, 9.1), List.of(true, 8.1)), result.get("values"));
+
+        // missing params
+        ResponseException re = expectThrows(
+            ResponseException.class,
+            () -> runEsqlSync(
+                requestObjectBuilder().query(
+                    format(
+                        null,
+                        "from {} | eval x1 = ?n1 | where ?n2 == x1 | stats xx2 = max(?n3) by ?n4 | keep ?n4, ?n5 | sort ?n4",
+                        testIndexName()
+                    )
+                ).params("[]")
+            )
+        );
+        String error = re.getMessage();
+        assertThat(error, containsString(": Unknown query parameter [n1]"));
+
+        // field name pattern is not supported in where/stats/sort/dissect/grok
+        re = expectThrows(
+            ResponseException.class,
+            () -> runEsqlSync(
+                requestObjectBuilder().query(format(null, "from {} | where ?n1 == 1", testIndexName()))
+                    .params("[{\"n1\" : {\"value\" : \"integer*\" , \"identifierpattern\" : true}}]")
+            )
+        );
+        error = re.getMessage();
+        assertThat(error, containsString("VerificationException"));
+        assertThat(error, containsString("Field name pattern [integer*] is not supported"));
+
+        re = expectThrows(
+            ResponseException.class,
+            () -> runEsqlSync(
+                requestObjectBuilder().query(format(null, "from {} | stats x = count(?n1)", testIndexName()))
+                    .params("[{\"n1\" : {\"value\" : \"integer*\" , \"identifierpattern\" : true}}]")
+            )
+        );
+        error = re.getMessage();
+        assertThat(error, containsString("VerificationException"));
+        assertThat(error, containsString("Field name pattern [integer*] is not supported"));
+
+        re = expectThrows(
+            ResponseException.class,
+            () -> runEsqlSync(
+                requestObjectBuilder().query(format(null, "from {} | sort ?n1", testIndexName()))
+                    .params("[{\"n1\" : {\"value\" : \"integer*\" , \"identifierpattern\" : true}}]")
+            )
+        );
+        error = re.getMessage();
+        assertThat(error, containsString("VerificationException"));
+        assertThat(error, containsString("Field name pattern [integer*] is not supported"));
+
+        re = expectThrows(
+            ResponseException.class,
+            () -> runEsqlSync(
+                requestObjectBuilder().query(format(null, "from {} | dissect ?n1 \"%{bar}\"", testIndexName()))
+                    .params("[{\"n1\" : {\"value\" : \"keyword*\" , \"identifierpattern\" : true}}]")
+            )
+        );
```

Comment:
Since the code is repetitive, this can be simplified by moving the arguments and error message in a List and then performing a loop.

Reply:
Done